### PR TITLE
Improvemets: test JSON RPC compatibility

### PR DESCRIPTION
**Pull request by koteld**

### CODE REVIEW

File: openrpc.json [hunk 1]
```diff
@@ -0,0 +1,4302 @@
+{
+	"openrpc": "1.2.4",
```
Comment (kalverra) [inline openrpc.json]: Can we move this to live in the folder with the tests? And maybe run it through a minimizer?

File: suite/compatibility/json_rpc_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+			}
+		})
+	})
+
+	Describe("Test GET RPC methods and validate results", func() {
+		It("OCR test GET Methods", func() {
+			for chainId, rpcClients := range rpcClientsByChain {
+				fmt.Printf("Starting tests for chain ID %d\n", chainId)
+				for rpcMethod, rpcMethodParameters := range rpcMethodCalls {
+					fmt.Printf("\n\nMethod: %s", rpcMethod)
```
Comment (kalverra) [inline suite/compatibility/json_rpc_test.go:98]: Make use of `zerolog` instead of the `fmt.Print`s.

File: suite/compatibility/json_rpc_test.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		}},
+	}
+
+	BeforeEach(func() {
+		By("Getting RPC clients", func() {
+			nc, err := config.LoadNetworksConfig(filepath.Join(utils.ProjectRoot, "networks.yaml"))
+			Expect(err).ShouldNot(HaveOccurred())
+			for _, networkName := range nc.SelectedNetworks {
+				networkSettings, ok := nc.NetworkSettings[networkName]
+				Equal(ok)
```
Comment (kalverra) [inline suite/compatibility/json_rpc_test.go:79]: It doesn't seem clear what this is checking or if it's really asserting anything. Does `Equal` run an assert on `ok` this way? Doesn't it need to have something to check what it's equal against?

File: suite/compatibility/json_rpc_test.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+						if value.Name == rpcMethod {
+							method = value
+							break
+						}
+					}
+					schemaLoader := gojsonschema.NewGoLoader(method.Result.Schema)
+					for _, rpcClient := range rpcClients {
+						var rpcCallResult interface{}
+						err := rpcClient.CallContext(context.Background(), &rpcCallResult, rpcMethod, rpcMethodParameters...)
+						if err != nil {
```
Comment (kalverra) [inline suite/compatibility/json_rpc_test.go:110]: Make use of the Ginkgo matchers and assertions instead.

**Review by kalverra — CHANGES_REQUESTED** [review]

File: openrpc.json (hunk 1 above)
Comment (koteld) [reply openrpc.json]: Yes, for sure, we can do that.
Also, we can try compiling this file from source just before the test: https://github.com/ethereum/execution-apis

File: suite/compatibility/json_rpc_test.go (hunk 3 above)
Comment (koteld) [reply suite/compatibility/json_rpc_test.go:79]: It's like assert.True(...), but I'll try to make it more obvious, or will remove this assertion.

Comment (kalverra) [reply suite/compatibility/json_rpc_test.go:79]: Ah okay, that's fine then. I wasn't aware of that behavior.

File: suite/compatibility/json_rpc_test.go (hunk 4 above)
Comment (koteld) [reply suite/compatibility/json_rpc_test.go:117]: I want test to continue so I don't use gomega assertions and use logging instead.
I'm not sure if it is reasonable to make separate tests for each JSON RPC method and use assertion for them.
If it is possible to continue test after failed assertion I will use it.

**Pull request closed without merging**